documents/data-sharing-annual-community-survey.md

```diff
@@ -0,0 +1,87 @@
+
+This initial report will be sent to all members of Rust teams, working groups, and the Rust Foundation. Recipients will be asked not to share the report outside the Rust Project and Foundation.
+
+The survey working group will present the report to the Rust Project and Foundation, and answer questions, for example in the Cross-Team Collaboration Fun Times meeting (note that this meeting is public and thus care must be taken not to discuss any sensitive information).
+
+With core team approval, this report will be shared publicly, but not strongly advertised (i.e., a post on the Inside Rust blog, but not the main blog).
+
+Unresolved question: is the core team the right group to approve the public report?
+
+## Ongoing reporting of data
+
+The survey working group will continue to analyse the survey data and may produce further reports, either for teams or the whole Rust Project. The scope of distribution and any requirements for confidentiality agreements will be decided on a case-by-case basis. These reports should never include PII. This responsibility may be transferred to a different Rust team or working group in the future.
+
+Teams and working groups may request further analysis of the survey data by the survey working group. The working group should aim to respond to all requests unless they would expose PII or allow for identification of respondents (e.g., by identifying a small cohort), or the workload on the group is too much. These requests will be prioritised by the survey working group. Responses to these requests may be shared with a wider group within the Rust Project, but should not be shared outside the Project or Foundation. The scope of distribution and any requirements for confidentiality agreements will be decided on a case-by-case basis.
+
+Unresolved question: should there be a time limit?
+
+## Raw data
+
+The full data set of survey responses will never be widely shared. It should be accessible only to the subset of the survey working group responsible for the annual community survey, contractors engaged to help process and analyse the data, and translators (who should only have access to the data they are translating, not the whole data set). These groups may be expanded in the future, but should be kept to a minimum. Everyone with access should have completed a signed confidentiality agreement with the Rust Foundation.
```

Comment:
The more we give access to raw data, the more we expose ourselves to the risk of individuals making honest mistakes and leaking data they shouldn't. I understand the desire to want to get out of folks way, but the "summary data" is still in a form that allows arbitrary querying. 

The only thing that the raw data allows for that summary data does not is analysis of free form text. While I can imagine insight coming from analyzing free from text without having a known goal ahead of time, this is a *lot* of work, and I don't imagine many individuals wanting to do this. If one is really that interested in the data, they should probably be on the survey team themselves and sharing their analysis with others.

Reply:
I agree that people are unlikely to want to "deep dive" into freeform data. That said, I've personally also found it useful to just read all the freeform responses from time to time, or at least big chunks of them, to get a sense for the "kinds of things" that people say.

Reply:
> Does summary data allow for correlating data across questions?

No, it doesn't, but that's a limitation of our current tooling not something that is limited by this policy. I think it makes more sense for us to fix the tooling to allow for this rather than opening up the data.

> to get a sense for the "kinds of things" that people say.

This is something the survey team would provide. The question becomes: is the risk associated with opening up data to others worth allowing others outside the survey team the ability to participate in this process in parallel?

Reply:
Perhaps "arbitrary querying" would not be allowed, but I can definitely imagine a system that allows correlating summary data as long as we limit queries to those that don't limit the data set beyond a certain point (i.e., don't allow queries that produce sets of less than ~500). 

Either way, the question at hand is whether we allow all Rust team members access to the data (provided a signed NDA) or not. I'd lean towards keeping raw data in the smallest set of hands possible, since the worst case scenario of leaking data is pretty bad (i.e., once data is leaked, it's leaked for good, and the only recourse we have is to enforce an NDA against a team member which sounds horrible). 

If we find that this set up is very painful, we can change the policy and go through the proper due diligence of informing survey respondents of this.

Reply:
I'm trying to decide how hard to push here. My thinking is like this: if we are going to seriously try to get use of the survey data (and I'm not sure whether it's the right tool for that, to be quite honest), I at least think the way I would do that is to sit down with the questions and read through them. I would find interesting questions that I'd like to understand better, and start formulating queries and formulations against the data to do that (e.g., people who said they are experts come from these domains, etc).

It's likely I would not want access to all the data. I would probably want access to "summary data" from free form comments ("of the known keywords, did this person reference it?"). I could imagine caring what country someone is from, but that seems fairly unlikely. 

In my *ideal* world, I would be doing these queries via some sort of web interface to a cool data analysis platform, so I would never actually have the data on my local machine. Thus the question of exporting would be pretty moot. 

In any case, that workflow could certainly work with survey team members as intermediaries, but it seems like a waste of everyone's time. And the problem is that it is hard for me to formulate queries up front, the choice of what to ask is maybe driven by the answers to previous questions.

Certainly I am influenced by this feeling that this data is -- by and large -- fairly impersonal, and so while I agree that leaking the whole data set would be bad, I feel like people's expectations of privacy are probably fairly low. I could be wrong in this assessment. 

All of that said: I would like to do the above for this year's data, I'm aware I didn't do that =)

Reply:
So, I think that there is a need for people to do that kind of deep dive into the data in general and that it would be beneficial to do. However, I think there is a trade-off here against professional treatment of people's data and that means having guard rails at least for how data is accessed and in particular minimising the number of people who have access to the raw data (even though the data is fairly impersonal, I think that there is still an expectation of privacy. I think part of being a 'good citizen' is that the Rust project treats the collected data with the highest level of respect for privacy and we follow best practices, etc even where that seems excessive. To give some concrete examples of hypothetical scenarios: 1, a respondent leaves a comment like "nrc is a big fat doodihead" in a free form answer. Given they come from a country with only 6 respondents and their answers to contribution questions indicate they are a regular contributor, nrc can identify them and is then (consciously or unconsciously) biased against them in future work. 2, 50% of respondents who speak Portuguese said they feel unwelcome due to their sexual orientation, this is mentioned as a joke by a team member at a meetup in Brazil and is overheard by someone with connections to a homophobic hate group, Rust meetups become targets for harassment.)

I do think that *someone* should do the kind of work you're describing and thus there is a role for that in this policy. Currently the survey WG, though I could imagine that being a product team in the future if such a thing ever existed. I strongly believe that it would be a mistake to expand access to the raw data to a large group of people (e.g., all team members or even all team leads). It is not unreasonable that we might want to expand the group beyond the survey WG, but I think that group should be well-defined, minimal, and have good reason to want to work with the data in this manner (i.e., curiosity is not a good enough reason). I don't have any suggestions for defining such a group, but I'd be happy to hear some.

Alternatively, you could join the survey WG to do this kind of analysis. I think expanding the number of people in the WG to analyse the data should be expanded, and that membership for the purpose of analysis without obligation to writing next year's survey, etc. is reasonable.

Reply:
I'm willing to let it stand as the survey wg (as it happens, I am currently a member anyway, but that's neither here nor there) and see how much trouble it is. My concern is that the result will be lower engagement with the data overall, but there are other ways to work on that. 

I still think the danger is minimal, particularly with some effort at education, and that team leads in particular (though also members) have clear good reasons to want to work with the data in this fashion (they are trying to decide what work to prioritize and how to direct their energies). That said, I'm  *not* sure how useful the data is in practice for that purpose, and in particular the correlated data (versus the simple summary). So I'd be happy to start with the summary and see where we go.